pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1815,7 +1816,7 @@ double round_trip(const char *p, char **q, char decimal, char sci, char tsep,
     double r = PyOS_string_to_double(p, q, 0);
     if (maybe_int != NULL) *maybe_int = 0;
     if (PyErr_Occurred() != NULL) *error = -1;
-    else if (r == Py_HUGE_VAL) *error = Py_HUGE_VAL;
+    else if (r == Py_HUGE_VAL) *error = (int)Py_HUGE_VAL;
```

Comment:
awkward code. Py_HUGE_VAL is a float and is used as an error code with  an implicit cast. made it explicit to silence the warning.

Reply:
Good find. Does the caller even use this?

Reply:
DIdn't check. Probably err !=0 at least.

Reply:
No worries. Makes sense to suppress build warning though as you can tell there are a lot of opportunities to really clean some of this up. If you say any can certainly push PRs

Resolving this one for now

Reply:
@pilkibun this part of the code gets less maintenance attention than other parts; is likely to have developed unused/unreachable branches.  Any of those you can identify and remove (separate PR(s)) will be welcome.

Reply:
i ph33r the parser code.